src/main/java/seedu/address/model/student/Student.java

```diff
@@ -185,13 +186,25 @@ public GenericSpecialisation getSpecialisation() {
 
     public void setSpecialisation(GenericSpecialisation specialisation) {
         this.specialisation = specialisation;
-        for (GraduationRequirement graduationRequirement : this.major.getDegreeProgramme()
+        try {
+            for (GraduationRequirement graduationRequirement : this.major.getDegreeProgramme()
                 .getGraduationRequirementList()) {
-            if (graduationRequirement instanceof FocusAreaGraduationRequirement) {
-                FocusAreaGraduationRequirement focusAreaGraduationRequirement =
-                        (FocusAreaGraduationRequirement) graduationRequirement;
-                focusAreaGraduationRequirement.setGenericSpecialisation(specialisation);
+                if (graduationRequirement instanceof CompoundGraduationRequirement) {
+                    for (GraduationRequirement graduationRequirement1 : ((CompoundGraduationRequirement) graduationRequirement).getGraduationRequirementList()) {
+                        if (graduationRequirement1 instanceof CompoundGraduationRequirement) {
+                            for (GraduationRequirement graduationRequirement2 : ((CompoundGraduationRequirement) graduationRequirement1).getGraduationRequirementList()) {
+                                if (graduationRequirement2 instanceof FocusAreaGraduationRequirement) {
+                                    FocusAreaGraduationRequirement focusAreaGraduationRequirement = (FocusAreaGraduationRequirement) graduationRequirement2;
+                                    focusAreaGraduationRequirement.setSpecialisation(specialisation);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
Could replace nested `if` statements with guard clauses to avoid arrowhead style code.

Reply:
@thetruevincentchow there are ```for``` loops between the ```if``` statements. Either way, I'll help Dan with making it linear